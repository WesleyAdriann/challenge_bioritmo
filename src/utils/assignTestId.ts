type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
