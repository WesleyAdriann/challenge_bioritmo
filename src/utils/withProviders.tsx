
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
