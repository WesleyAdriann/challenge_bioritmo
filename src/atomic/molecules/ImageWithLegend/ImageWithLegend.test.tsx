import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'
import maskRecommended from '../../../assets/images/mask/recommended-mask.png'

import { ImageWithLegend, IImageWithLegend } from './index'

describe('Organisms/FormSelectHour', () => {
  const props: IImageWithLegend = {
    testID: 'ImageWithLegend',
    legend: 'Legend test',
    image: maskRecommended
  }

  const renderComponent = () =>
    render(<ImageWithLegend {...props} />, { wrapper })

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('div', props.testID!)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
