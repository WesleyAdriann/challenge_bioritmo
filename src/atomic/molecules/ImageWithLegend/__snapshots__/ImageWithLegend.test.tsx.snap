// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/ImageWithLegend should render snapshot 1`] = `
.c2 {
  color: #000;
  font-weight: 400;
  font-size: 14px;
  text-align: left;
  text-transform: none;
  -webkit-user-select: initial;
  -moz-user-select: initial;
  -ms-user-select: initial;
  user-select: initial;
  display: block;
  line-height: normal;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  height: 60px;
  margin-bottom: 8px;
}

<div
  class=" c0"
  data-testid="div_ImageWithLegend"
>
  <img
    class="c1"
    src="recommended-mask.png"
  />
  <p
    class="c2"
    data-testid="text_Text"
  >
    Legend test
  </p>
</div>
`;
